### PR TITLE
More updates for your 2.4.2 pull request

### PR DESCRIPTION
**Pull request by idlechild**
I pulled in cout's changes.  Also fixed some stuff mentioned in practice-hack channel:
- I busted elevator CF a few months back
- Segment timer resets at end of MB2/MB3 preset, I changed to reset on next event which is conveniently end of MB1 fight
- Added bomb spread to quick drop strat (I thought I did that already, not sure why I didn't, it was easy)

Thanks for finding my mistake with 41/43 variables, but I really did want to use 41/43 so when attempting to load C1/C3 in memory viewer so we can draw it... we can't use C1/C3 to load C1/C3!  Thus wanted to use 41/43 there.  Hopefully I got it right this time.

I need to review my infohud preservation during spacetime beam one more time.  I think it works but the math is tricky.  Also need to remember to add protection for xray.  Otherwise I think we are getting close to releasing 2.4.2.

Hey whatever happened to the infinite enemy fight logic?  Something like that might be useful for low ice training, I was thinking of an option to create a 400 or 800 low ice fight instead of the 200 shot fight.

### CODE REVIEW

**Comment by InsaneFirebat** [comment]
lol I just gave my pull a final review and marked it ready, now I see this.

The segment timer thing with MB presets is unfortunate. I much prefer it to not include any of the door transition after the preset is loaded. Can we have the MB preset logic unset the "reset-later" flag instead?

You may be right about spacetime. I did run into one issue that I suspected the spacetime protection of missing, but I was streaming and didn't test it.

**Comment by idlechild** [comment]
I think reset later is fine; I just changed "later" mean after door transition or any time ih_hud_code_early is called.  In this case MB1 death, which happens almost immediately upon loading an MB2/MB3/Zebes Escape preset.

**Comment by InsaneFirebat** [comment]
Resetting at `ih_hud_code_early` is what I wanted to avoid. This makes it reset on elevators  as well. If we add `LDA #$0000 : STA !ram_reset_segment_later` to `.set_mb_state`, it achieves the same thing without affecting timing elsewhere.

**Comment by InsaneFirebat** [comment]
> Hey whatever happened to the infinite enemy fight logic? Something like that might be useful for low ice training,

Added this to my pull.

**Comment by idlechild** [comment]
What's wrong with having it reset on elevators?  It will still only reset if the flag is set after preset load or user choice.

**Comment by InsaneFirebat** [comment]
It just keeps timing the same for people that track segments. DarkXoa and I have spreadsheets to track the stats and about half of our segments start on an elevator after the transition. We can still make it work the same, just need to menu after taking the elevator.

**Comment by idlechild** [comment]
Made change to set_mb_state.  I just need to test this out some more and I think 2.4.2 update will be ready to go.

**Comment by InsaneFirebat** [comment]
Were you still planning to do something with the 400/800 shot fights? We could LSR the damage value in the pacifist hijacks pretty easily. That also got me thinking about a room strat that counts how many beam shots are left to kill Ridley. It could also display the "death grab" counter once Ridley is at zero HP.

File: src/infohud.asm [hunk 1]
```diff
@@ -1471,5 +1482,5 @@ HexToNumberGFX2:
     dw #$0C09, #$0C00, #$0C01, #$0C02, #$0C03, #$0C04, #$0C05, #$0C06, #$0C07, #$0C08
 
 print pc, " infohud bank80 end"
-warnpc $80FFC0 ; header
+warnpc $80FFB0 ; header
```
Comment (InsaneFirebat) [inline src/infohud.asm:1485]: SM starts its game header at $80FFC0, with the game name. I skips the first half of the registration data and fills it with $FF.

Comment (idlechild) [reply src/infohud.asm:1485]: SM game header starts at FFC0, but there is the extended header starting at FFB0, figure we should stay out of that region.

**Comment by idlechild** [comment]
> 

We could but effort.  I've been less motivated to work on the practice rom lately; I'm sure the motivation will return I just need time.

Laziness not always bad.  Having a 400/800 shot option and settings to make MB2 more or less trolly (e.g. more onion rings) would be neat... but when you think about it, pacifist mode works.  I can (hopefully) get just as good at a low ice fight with pacifist mode and without all the other options.

**Comment by InsaneFirebat** [comment]
The only problem with Pacifist mode is that you're stuck with whatever boss HP you turn it on at. Ridley will never turn red and speed up if you don't turn it off and deal some real damage.

I could take some time off from new changes and give you a break. I appreciate you keeping up with my work on top of yours.

**Comment by idlechild** [comment]
It's easy enough to toggle the pacifist mode if you want to deal some damage for Ridley and/or MB

Lol you can do whatever schedule, I don't mind keeping up with your changes.

**Comment by InsaneFirebat** [comment]
Moved to https://github.com/tewtal/sm_practice_hack/pull/86

**Pull request closed without merging**